etc/cmake/options.cmake

```diff
@@ -34,6 +34,8 @@ option(OT_FTD "enable FTD" ON)
 option(OT_MTD "enable MTD" ON)
 option(OT_RCP "enable RCP" ON)
 
+option(OT_TCP "enable TCP" ON)
```

Comment:
Insert this into the list of `OT_*` options below (in alphabetical order).

Reply:
Do we want this to also control `OPENTHREAD_CONFIG_TCP_ENABLE`?
https://github.com/openthread/openthread/blob/6bfcc0d7d495fcc7bcf1cdd4070bee14c391ef95/src/core/config/ip6.h#L172

If so, we probably want to use `ot_option` instead.